week-5/Homework/mandatory/2-project/js/main.js

```diff
@@ -0,0 +1,97 @@
+//settings for blue button
+
+let blueBtn = document.querySelector("#blueBtn");
+blueBtn.addEventListener("click", blueBtnFunc);
+
+function blueBtnFunc() {
+  document.querySelector(".jumbotron").style.backgroundColor = `#588fbd`;
+  document.querySelector(
+    ".btn.btn-primary.btn-lrg"
+  ).style.backgroundColor = `#ffa500`;
+  document.querySelector(
+    ".btn.btn-secondary.btn-lrg"
+  ).style.backgroundColor = `black`;
+  document.querySelector(".btn.btn-secondary.btn-lrg").style.color = `white`;
+}
+
+//settings for orange button
+
+let orangeBtn = document.querySelector("#orangeBtn");
+orangeBtn.addEventListener("click", orangeBtnFunc);
+
+function orangeBtnFunc() {
+  document.querySelector(".jumbotron").style.backgroundColor = `#f0ad4e`;
+  document.querySelector(
+    ".btn.btn-primary.btn-lrg"
+  ).style.backgroundColor = `#5751fd`;
+  document.querySelector(
+    ".btn.btn-secondary.btn-lrg"
+  ).style.backgroundColor = `#31b0d5`;
+  document.querySelector(".btn.btn-secondary.btn-lrg").style.color = `white`;
+}
+
+//settings for green button
+
+let greenBtn = document.querySelector("#greenBtn");
+greenBtn.addEventListener("click", greenBtnFunc);
+
+function greenBtnFunc() {
+  document.querySelector(".jumbotron").style.backgroundColor = `#87ca8a`;
+  document.querySelector(
+    ".btn.btn-primary.btn-lrg"
+  ).style.backgroundColor = `black`;
+  document.querySelector(
+    ".btn.btn-secondary.btn-lrg"
+  ).style.backgroundColor = `#8c9c08`;
+}
+
+//settings for submit button
+
+let submitBtn = document.querySelector(".form-group + .btn.btn-primary");
+submitBtn.addEventListener("click", submitFunc);
+
+function submitFunc() {
+  let email = document.querySelector("#exampleInputEmail1");
+  let yourName = document.querySelector("#example-text-input");
+  let describeYourself = document.querySelector("#exampleTextarea");
+
+  if (
+    email.value.length === 0 ||
+    yourName.value.length === 0 ||
+    describeYourself.value.length === 0 ||
+    email.value.includes("@") === false
+  ) {
+    if (email.value.length === 0 || email.value.includes("@") === false) {
+      email.style.backgroundColor = "red";
+    }
+    if (yourName.value.length === 0) {
+      yourName.style.backgroundColor = "red";
+    }
+    if (describeYourself.value.length === 0) {
+      describeYourself.style.backgroundColor = "red";
+    }
+    event.preventDefault();
+  } else {
+    alert("Thank you for filling");
+    event.preventDefault();
+    email.style.backgroundColor = "white";
+    yourName.style.backgroundColor = "white";
+    describeYourself.style.backgroundColor = "white";
+    email.value = "";
+    yourName.value = "";
+    describeYourself.value = "";
+  }
+}
+
```

Comment:
This works, but it's quite complicated. There might well be ways you could simplify it to make it easier to maintain, for example so as you don't have to check any of the conditions twice. In this situation it doesn't matter much, but it's possible you might be doing something in another situation where you have to run a function that takes a while, or accesses network resources, which you'd want to keep to a minimum.

Reply:
Thanks for your feedback. I changed the code and removed dual check-up. But I noticed that code is working differently. It paints in red invalid fields one after another, not simultaneously, as it was early. I am not sure that it is so bad, but it can increase quantity of button pressures. Thanks for your assistance.
`function submitFunc() {
  let email = document.querySelector("#exampleInputEmail1");
  let yourName = document.querySelector("#example-text-input");
  let describeYourself = document.querySelector("#exampleTextarea");
  event.preventDefault();

  if (email.value.length === 0 || email.value.includes("@") === false) {
    email.style.backgroundColor = "red";
  } else if (yourName.value.length === 0) {
    yourName.style.backgroundColor = "red";
  } else if (describeYourself.value.length === 0) {
    describeYourself.style.backgroundColor = "red";
  } else {
    alert("Thank you for filling");
    email.style.backgroundColor = "white";
    yourName.style.backgroundColor = "white";
    describeYourself.style.backgroundColor = "white";
    email.value = "";
    yourName.value = "";
    describeYourself.value = "";
  }
}`